snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java

```diff
@@ -203,18 +201,34 @@ private Boolean persistChanges(final BranchContext context, final IProgressMonit
 		final BulkRequestBuilder<TransactionContext> bulkRequestBuilder = BulkRequest.create();
 
 		applyChanges(subMonitor, context, bulkRequestBuilder);
-		final Request<BranchContext, CommitResult> commitRequest = SnomedRequests.prepareCommit()
-			.setBody(bulkRequestBuilder)
-			.setCommitComment(commitComment)
-			.setParentContextDescription(DatastoreLockContextDescriptions.SAVE_CLASSIFICATION_RESULTS)
-			.setAuthor(userId)
-			.build();
-
-		final CommitResult commitResult = new IdRequest<>(commitRequest)  
-			.execute(context);
-
-		classificationTracker.classificationSaved(classificationId, commitResult.getCommitTimestamp());
-		return Boolean.TRUE;
+	
+		long resultTimeStamp = Commit.NO_COMMIT_TIMESTAMP;
+		for (List<Request<TransactionContext, ?>> partition : Iterables.partition(bulkRequestBuilder.build().getRequests(), getCommitLimit(context))) {
```

Comment:
One additional (future) idea: add a method to `BulkRequest` that breaks itself into smaller `BulkRequest`s of a certain size, either up front as a `List`, or as a lazily-computed `Iterable` or `Stream`. 😃